error-prone-contrib/src/test/java/tech/picnic/errorprone/bugpatterns/CanonicalConstantFieldNameTest.java

```diff
@@ -0,0 +1,156 @@
+            "    int number = 3;",
+            "",
+            "    return number;",
+            "  }",
+            "",
+            "  class B {",
+            "    private static final int number = 4;",
+            "",
+            "    private static final int numberFromAnotherClass = 5;",
+            "  }",
+            "}")
+        .addOutputLines(
+            "A.java",
+            "class A {",
+            "  private static final int NUMBER = 1;",
+            "",
+            "  int nonConstantNumber = 2;",
+            "",
+            "  int referenceToNumberFromAnotherClass = B.NUMBER_FROM_ANOTHER_CLASS;",
+            "",
+            "  static int getConstantNumber() {",
+            "    return NUMBER;",
+            "  }",
+            "",
+            "  static int getLocalNumber() {",
+            "    int number = 3;",
+            "",
+            "    return number;",
+            "  }",
+            "",
+            "  class B {",
+            "    private static final int NUMBER = 4;",
+            "",
+            "    private static final int NUMBER_FROM_ANOTHER_CLASS = 5;",
+            "  }",
+            "}")
+        .doTest(TestMode.TEXT_MATCH);
+  }
+
+  @Test
+  void doNotReplaceExcludedOrPublicConstantsByDefault() {
+    BugCheckerRefactoringTestHelper.newInstance(CanonicalConstantFieldName.class, getClass())
+        .addInputLines(
+            "A.java",
+            "class A {",
+            "  private static final long serialVersionUID = 1L;",
+            "",
+            "  public static final int number = 1;",
+            "",
+            "  static final int anotherNumber = 2;",
+            "}")
+        .addOutputLines(
+            "A.java",
+            "class A {",
+            "  private static final long serialVersionUID = 1L;",
+            "",
+            "  public static final int number = 1;",
+            "",
+            "  static final int anotherNumber = 2;",
+            "}")
+        .doTest(TestMode.TEXT_MATCH);
+  }
+
+  @Test
+  void excludeFlaggedConstants() {
+    BugCheckerRefactoringTestHelper.newInstance(CanonicalConstantFieldName.class, getClass())
+        .setArgs("-XepOpt:CanonicalConstantFieldName:ExcludedConstantFieldNames=foo")
+        .addInputLines(
+            "A.java",
+            "class A {",
+            "  private static final int number = 1;",
+            "",
+            "  private static final int foo = 3;",
+            "}")
+        .addOutputLines(
+            "A.java",
+            "class A {",
+            "  private static final int NUMBER = 1;",
+            "",
+            "  private static final int foo = 3;",
+            "}")
+        .doTest(TestMode.TEXT_MATCH);
+  }
+
+  @Test
+  void includePublicConstants() {
+    BugCheckerRefactoringTestHelper.newInstance(CanonicalConstantFieldName.class, getClass())
+        .setArgs("-XepOpt:CanonicalConstantFieldName:IncludePublicConstantFields=true")
```

Comment:
Hmm, interesting. This can lead to quite some unexpected situations right 👀 ? As we cannot rename usages of this in other files, so I'm not sure if we should do this 🤔.

Reply:
We have a _replacement_ test for this situation; where we rename a variabled referenced in another file. 
See
```
  int referenceToNumberFromAnotherClass = B.numberFromAnotherClass;
```

Reply:
I have a test for this where we replace with suggested fix the variable imported from another class.